src/Purestone/Card.hs

```diff
@@ -27,6 +30,9 @@ data Card = Card {
     cardParadigms :: NonEmpty Paradigm,
     -- | The card's program.
     cardProgram :: Program
-} deriving (Eq, Show)
+} deriving (Eq, Show, Read, Generic)
+
+instance FromJSON Card
+instance ToJSON Card
```

Comment:
Have you checked that the generated JSON is of a desirable format (naming-wise)? I normally write the instances by hand to get the naming I want, so I am unsure what the generic versions generate by default.

Reply:
It uses the record field names for the keys. If we want something else I'll write the instances manually

Reply:
I think that is generally undesirable, since the prefixes in the record names are really not necessary. You could do something similar to https://github.com/mbg/uow-apis/blob/master/src/Warwick/Tabula/JSON.hs#L37 and then you can have e.g.
```haskell
instance FromJSON Submission where
    parseJSON = parseTabulaJSON
```

Reply:
There is also http://hackage.haskell.org/package/deriving-aeson

Reply:
Which method would be preferable

Reply:
I am not sure. I have not used `deriving-aeson` before, since it is a relatively recent library (I just read about it the other day). It adds a new dependency and requires some more language extensions (I am not opposed to either, though). The standard `aeson` method is conceptually simpler and doesn't require anything new. I will leave it up to you to decide!

Reply:
👍 I will have a closer look at both methods and implement one in the morning